pkg/clusterd/disk.go

```diff
@@ -213,3 +215,11 @@ func PopulateDeviceUdevInfo(d string, executor exec.Executor, disk *sys.LocalDis
 
 	return disk, nil
 }
+
+func SetAllowLoopDevices(val bool) {
+	allowLoopDevices = val
```

Comment:
Thank you for your comment.

`SetAllowLoopDevices` and `LoopDevicesAllowed` functions are used for two kinds of processes.
1. Operator
2. Pods (prepare job, discover daemonset, and cleanup job)

As for the second one, `SetAllowLoopDevices` was called at the beginning of each pod, and then `LoopDevicesAllowed` was used at `supportedDeviceType` to check if loop devices are allowed. I could eliminate the setter/getter call by introducing `GetAllowLoopDevices()` as you suggested.

However, for the operator, it should create each pod based on the latest `ROOK_CEPH_ALLOW_LOOP_DEVICES` parameter. So we need some means to know the value of it when creating these pods (at `pkg/operator/ceph/cluster/cleanup.go`, `pkg/operator/ceph/cluster/osd/provision_spec.go`, and `pkg/operator/discover/discover.go`).

What do you think is good for that? The following list is what I came up with.

- Use `SetAllowLoopDevices` and `LoopDevicesAllowed` as it is.
  - In that case, it will be better to move them to some other file since it is no longer used by `supportedDeviceType()`.
- Define a new environment variable for the operator process, and use it to propagate the current value of `ROOK_CEPH_ALLOW_LOOP_DEVICES` parameter.
- Do not check the value of `ROOK_CEPH_ALLOW_LOOP_DEVICES` parameter at `operator.reconcile()`. Instead, check the value directory from the `rook-ceph-operator-config` ConfigMap when creating each pod.

Reply:
> Define a new environment variable for the operator process, and use it to propagate the current value of ROOK_CEPH_ALLOW_LOOP_DEVICES parameter.

For now, I adopted this approach and declared a new env `ROOK_CEPH_ALLOW_LOOP_DEVICES`, which has the same name as the new ConfigMap parameter.